pkg/core/transaction/transaction.go

```diff
@@ -119,7 +130,21 @@ func (t *Transaction) DecodeBinary(br *io.BinReader) {
 	t.Version = uint8(br.ReadB())
 	t.Nonce = br.ReadU32LE()
 	t.Sender.DecodeBinary(br)
-
+	t.SystemFee.DecodeBinary(br)
+	if t.SystemFee < 0 {
+		br.Err = errors.New("negative system fee")
+		return
+	}
+	t.NetworkFee.DecodeBinary(br)
+	if t.NetworkFee < 0 {
+		br.Err = errors.New("negative network fee")
+		return
+	}
+	// something strange: https://github.com/neo-project/neo/blob/master/src/neo/Network/P2P/Payloads/Transaction.cs#L168
+	if t.NetworkFee+t.SystemFee < t.SystemFee {
+		br.Err = errors.New("negative network fee")
```

Comment:
It's an overflow check, so there is nothing negative about these fees, it's just that they're too big to be correctly processed using int64.

Reply:
Got it.